cognite/extractorutils/unstable/core/_dto.py

```diff
@@ -30,17 +33,77 @@ def dict(self, *args: Any, **kwargs: Any) -> dict[str, Any]:  # noqa: ANN401
     model_config = ConfigDict(alias_generator=camelize, populate_by_name=True, extra="forbid")
 
 
+class WithExternalId(CogniteModel):
+    external_id: str
+
+
+MessageType = Annotated[str, StringConstraints(min_length=0, max_length=1000)]
+
+
 class TaskUpdate(CogniteModel):
     type: Literal["started"] | Literal["ended"]
     name: str
     timestamp: int
+    message: MessageType | None = None
 
 
-class Error(CogniteModel):
-    external_id: str
+class Error(WithExternalId):
     level: str
     description: str
     details: str | None
     start_time: int
     end_time: int | None
     task: str | None
+
+
+TaskUpdateList = Annotated[list[TaskUpdate], Len(min_length=1, max_length=1000)]
+ErrorList = Annotated[list[Error], Len(min_length=0, max_length=1000)]
+VersionType = Annotated[str, StringConstraints(min_length=1, max_length=32)]
+DescriptionType = Annotated[str, StringConstraints(min_length=0, max_length=500)]
+TaskList = Annotated[list["Task"], Len(min_length=1, max_length=1000)]
+JSONType = TypeAliasType(  # type: ignore
+    "JSONType",
+    bool
+    | int
+    | float
+    | str
+    | None
+    | list[Optional["JSONType"]]  # type: ignore
+    | dict[str, Optional["JSONType"]],  # type: ignore  # type: ignore
+)
```

Comment:
For these type-ignores it's good practice to use `# type: ignore[specific-error-mode]`, both to indicate what is being ignored and also to avoid unintentionally ignoring different error modes.